nexus/reconfigurator/planning/src/blueprint_builder/zones.rs

```diff
@@ -71,6 +72,31 @@ impl BuilderZonesConfig {
         Ok(())
     }
 
+    pub(super) fn expunge_zone(
+        &mut self,
+        zone_id: OmicronZoneUuid,
+    ) -> Result<(), BuilderZonesConfigError> {
+        let zone = self
+            .zones
+            .iter_mut()
+            .find(|zone| zone.zone.id == zone_id)
+            .ok_or_else(|| {
+            let mut unmatched = BTreeSet::new();
+            unmatched.insert(zone_id);
+            BuilderZonesConfigError::ExpungeUnmatchedZones { unmatched }
```

Comment:
rustfmt seems asleep at the switch here.

Reply:
I don't think so? If I change the formatting on this line, rustfmt puts it back to the way it is.

Reply:
... but it's definitely wrong, right?  I would expect L84-86 to be indented because they're inside a block that ends at line 87.  I'd also expect L87 to be indented because it closes the block and parens opened at L83.

Reply:
It's definitely weird and _probably_ wrong, and possibly related to our `rustfmt.toml` settings:

```toml
max_width = 80
use_small_heuristics = "max"
```

If I comment out `use_small_heuristics`, it doesn't change. If I comment out both lines, I get what you want:

```rust
        let zone = self
            .zones
            .iter_mut()
            .find(|zone| zone.zone.id == zone_id)
            .ok_or_else(|| {
                let mut unmatched = BTreeSet::new();
                unmatched.insert(zone_id);
                BuilderZonesConfigError::ExpungeUnmatchedZones { unmatched }
            })?;
```

If I comment out just `max_width` and keep `use_small_heuristics`, I get this (which also seems reasonable and more correct than what we're getting now):

```rust
        let zone = self.zones.iter_mut().find(|zone| zone.zone.id == zone_id).ok_or_else(|| {
            let mut unmatched = BTreeSet::new();
            unmatched.insert(zone_id);
            BuilderZonesConfigError::ExpungeUnmatchedZones { unmatched }
        })?;
```

I think I will just leave this alone - I'm not sure how to isolate it, or whether it's a rustfmt bug or just some interaction we don't understand.